[Team] LNB, TeamSection 컴포넌트 초안

ingbinsee opened the pull request:
## 🔍️ 이 PR을 통해 해결하려는 문제가 무엇인가요?

> 어떤 기능을 구현한건지, 이슈 대응이라면 어떤 이슈인지 PR이 열리게 된 계기와 목적을 Reviewer 들이 쉽게 이해할 수 있도록 적어 주세요
> 일감 백로그 링크나 다이어그램, 피그마를 첨부해도 좋아요

- Team 페이지의 기능 및 레이아웃(공통 section 제외)을 구현하였습니다.

## ✨ 이 PR에서 핵심적으로 변경된 사항은 무엇일까요?

> 문제를 해결하면서 주요하게 변경된 사항들을 적어 주세요

- 분기점(모바일, 태블릿, 데스크탑)에 따른 반응형 레이아웃 구현
- Lnb, TeamSection 컴포넌트 제작
- teamData.js의 데이터를 filter 함수를 활용하여 렌더링

https://github.com/ReOpenSG/ReOpenSG/assets/140426866/d527b3a4-cffe-40b1-86ed-86f1b25e5bbe

## 🔖 핵심 변경 사항 외에 추가적으로 변경된 부분이 있나요?

> 없으면 "없음" 이라고 기재해 주세요

- 없음

## 🙏 Reviewer 분들이 이런 부분을 신경써서 봐 주시면 좋겠어요

> 개발 과정에서 다른 분들의 의견은 어떠한지 궁금했거나 크로스 체크가 필요하다고 느껴진 코드가 있다면 남겨주세요

- 없음

## 🩺 이 PR에서 테스트 혹은 검증이 필요한 부분이 있을까요?

> 테스트가 필요한 항목이나 테스트 코드가 추가되었다면 함께 적어주세요

- 없음

### 📌 PR 진행 시 이러한 점들을 참고해 주세요

- Reviewer 분들은 코드 리뷰 시 좋은 코드의 방향을 제시하되, 코드 수정을 강제하지 말아 주세요.
- Reviewer 분들은 좋은 코드를 발견한 경우, 칭찬과 격려를 아끼지 말아 주세요.
- Review는 특수한 케이스가 아니면 Reviewer로 지정된 시점 기준으로 3일 이내에 진행해 주세요.
- Comment 작성 시 Prefix로 P1, P2, P3 를 적어 주시면 Assignee가 보다 명확하게 Comment에 대해 대응할 수 있어요
  - P1 : 꼭 반영해 주세요 (Request Changes) - 이슈가 발생하거나 취약점이 발견되는 케이스 등
  - P2 : 반영을 적극적으로 고려해 주시면 좋을 것 같아요 (Comment)
  - P3 : 이런 방법도 있을 것 같아요~ 등의 사소한 의견입니다 (Chore)

#

---

## 📝 Assignee를 위한 CheckList

- [ ] To-Do Item

ingbinsee committed:
🗑️ [Deprecated] : 중복 코드 삭제

중복으로 불러온 'react-toastify' 1줄을 삭제하였습니다.

ingbinsee committed:
⚙️ [Chore] : blue 색상 삭제 및 accent 색상 변경

blue-50 ~ blue-950을 삭제하였으며, accent 색상을 #0071E3으로 지정하였습니다.
색상 변경에 따라 발생한 코드 오류를 해결하였습니다.
+ Admin.module.css의 blue-900을 accent-accent로 변경하였습니다.
+ Contact.module.css의 blue-900을 accent-accent로 변경하였습니다.

ingbinsee committed:
⚙️ [Chore] : 공통 환경 세팅 변경

eslint 일부 옵션을 off하였습니다.

ingbinsee committed:
✨ [Feat] : LNB 컴포넌트 완료

LNB 컴포넌트를 반응형으로 구현하였습니다.
+ 부서, 팀, 직무 등의 내용을 포함하는 teamData.js를 구성하였습니다.
+ 데이터를 불러오는 방식을 차용하여 LNB 요소를 렌더링하였습니다.
+ LNB 요소 클릭 시 해당 text를 포함한 사이트 주소로 변경될 수 있도록 하였습니다.
+ 분기점(모바일, 태블릿, 데스크탑)에 따른 변화를 반영하였습니다.
+ module.css로 스타일을 분리하였습니다.

Related to: #58

ingbinsee committed:
✨ [Feat] : TeamSection 컴포넌트 완료
TeamSection 컴포넌트를 반응형으로 구현하였습니다.
+ LNB 클릭 시 teamData.js를 활용하여 해당하는 내용의 데이터를 렌더링하도록 하였습니다.
+ LNB 클릭 시 선택된 LNB의 색상이 변경되도록 하였습니다.
+ 아이콘을 컴포넌트로 제작하여 분기점별 크기가 변경되도록 하였습니다.
+ 분기점(모바일, 태블릿, 데스크탑)에 따른 변화를 반영하였습니다.

Related to: #59

ingbinsee committed:
🤝 [Merge] : 원격 저장소와 로컬 저장소 동기화

ingbinsee committed:
🌈 [Design] : 색상 변경 및 버튼 스타일링

색상 변경에 따라 발생한 코드 오류를 해결하였습니다.
+ Contact.jsx의 blue-900을 accent-accent로 변경하였습니다.
+ InquiryInput.jsx의 blue-900을 accent-accent로 변경하였습니다.
+ MachinesDevices.module.css의 blue-900을 accent-accent로 변경하였습니다.

<li>에 반영되어 있던 스타일링을 <button>에 적용하였습니다.
hover 시 DownloadCard 컴포넌트의 배경 색상이 변하는 것에서 border 색상이 생기는 것으로 변경하였습니다.

dani-day commented on src/components/Team/TeamLnb.jsx:
```diff
@@ -0,0 +1,48 @@
+  return (
+    <ul className={styles.lnbUl}>
+      <li>
+        <button
+          type="button"
+          id={id}
+          onClick={() => {
+            window.location.pathname = '/team';
+          }}
+          className={
+            id.id === undefined ? '-bg--open-accent-accent' : '-bg--openfoundation-quaternary'
+          }
+        >
+          전체
+        </button>
+      </li>
+      {data.map((list) => (
+        <li>
+          <button
+            type="button"
+            id={id}
+            onClick={() => {
+              window.location.pathname = `/team/${list[0]}`;
+            }}
+            className={
+              list[0] === id.id ? '-bg--open-accent-accent' : '-bg--openfoundation-quaternary'
+            }
+          >
+            {list[0]}
+          </button>
```
P1: 버튼 클릭 시, 부분 렌더링이 아니라 전체 페이지 이동으로 작동하는 것 같습니다. 확인 후 수정 부탁드리겠습니다. :)

ingbinsee replied:
수정하였습니다.

dani-day commented on src/views/Team.jsx:
```diff
@@ -1,7 +1,29 @@
+      /> */}
+      <div className={styles.teamContainer}>
+        <div className={styles.sectionWrapper}>
+          <TeamLnb data={data} />
```
P1: 직무 버튼 클릭 시, 콘솔창에 아래 Warning 이 발생하는 것 같습니다. 에러 해결해주시면 좋을 것 같습니다.

![image](https://github.com/ReOpenSG/ReOpenSG/assets/134567470/da57f3ce-eb86-4aa0-acc0-b7f16f97c4aa)

ingbinsee replied:
해결하였습니다.

dani-day commented on src/components/Team/TeamLnb.jsx:
```diff
@@ -0,0 +1,48 @@
+function TeamLnb({ data }) {
+  const id = useParams();
+  return (
+    <ul className={styles.lnbUl}>
```
P1: 아래 콘솔창 warning 해결해주시면 좋을 것 같습니다.

![image](https://github.com/ReOpenSG/ReOpenSG/assets/134567470/43d6c739-5ce8-45e3-9a0a-67bf3b0b2dd6)

ingbinsee replied:
해결하였습니다.

dani-day commented on src/components/Team/RoleSection.jsx:
```diff
@@ -0,0 +1,34 @@
+
+  return (
+    <ul>
+      <li className={styles.sectionWrapper}>
```
P1: 아래 콘솔창 warning 해결해주시면 좋을 것 같습니다.

![image](https://github.com/ReOpenSG/ReOpenSG/assets/134567470/c94146af-6def-4465-ab95-1720720355e2)

ingbinsee replied:
해결하였습니다.

dani-day reviewed (changes requested):
수고하셨습니다 👍  P1 review 4개 남겨두었습니다. 콘솔창 에러가 있어 부득이하게 P1 으로 달아두었으니 확인 부탁드리겠습니다.

ingbinsee committed:
⚙️ [Chore] : 공통 환경 세팅 변경

eslint 일부 옵션을 off하였습니다.

ryudg commented on src/components/Team/RoleSection.jsx:
```diff
@@ -0,0 +1,34 @@
+    data.filter((list) => (list[0] === id.id ? setContent(list[1]) : ''));
+  });
+
+  useEffect(() => {});
```
**P3**: 불필요 코드 삭제 부탁드립니다.

ingbinsee replied:
삭제하였습니다.

ingbinsee committed:
♻️ [Refactor] : 리뷰 사항 반영

불필요한 페이지 이동 대신 버튼 클릭 시 직무 데이터만 변경되도록 수정하였습니다.
console창에 있는 에러 메시지를 해결하였습니다.

Related to: #73

kelvin-chihyun commented on src/components/Team/TeamLnb.jsx:
```diff
@@ -0,0 +1,50 @@
+          전체
+        </button>
+      </li>
+      {data.map((list) => (
```
P1: map 메서드 사용시 key prop은 권장사항이라 작성해주시면 좋을 것 같습니다. 콘솔에 에러가 뜨네요!!

ingbinsee replied:
해결하였습니다.

ryudg commented on src/components/Team/RoleSection.jsx:
```diff
@@ -0,0 +1,33 @@
+    <ul>
+      <li className={styles.sectionWrapper}>
+        {state === '전체' || state === '' ? <RoleAllSection /> : ''}
+        {content?.map((item) => (
```
P3: 부모 컴포넌트인 Team.jsx의 state default 값을 '전체'라고 설정한다면, `{ state === '전체' && <RoleAllSection /> }`와 같이 코드를 줄여 사용할 수 있습니다.

ingbinsee replied:
감사합니다. 코드가 깔끔해졌습니다.
수정하였습니다.

ryudg commented on src/components/Team/TeamLnb.jsx:
```diff
@@ -0,0 +1,50 @@
+              ? '-bg--open-accent-accent'
+              : '-bg--openfoundation-quaternary'
+          }
+        >
```
P3: isClicked라는 클래스를 추가해 작업하면 중복 코드를 제거할 수 있습니다.
```css
  & button {
    @apply px-open-xl tablet:px-open-2xl py-open-sm rounded-[20px] -bg--openfoundation-quaternary;
    
    &.isClicked {
      @apply -bg--open-accent-accent;
    }
  }
```

ingbinsee replied:
중복된 코드는 이렇게 하면 반복하여 사용하지 않아도 조건이 적용되는군요. 감사합니다.
수정하였습니다.

ryudg reviewed (approved):
`src/components/Team/RoleSection.jsx`, `src/components/Team/TeamLnb.jsx` 코드에 comment 추가했습니다.

ingbinsee committed:
🗑️ [Deprecated] : 불필요한 코드 삭제

사용하지 않는 코드(useEffect)를 삭제하였습니다.

Related to: #73

ingbinsee committed:
♻️ [Refactor] : 코드 단순화

Team.jsx의 state default 값을 빈 문자에서 '전체'로 변경하였습니다.
{state === '전체' || state === '' ? <RoleAllSection /> : ''} 코드를 { state === '전체' && <RoleAllSection /> }와 같이 줄여 사용하였습니다.

Related to: #73

ingbinsee committed:
♻️ [Refactor] : 중복된 코드 사용 개선

isClicked라는 클래스를 추가하여 조건이 성립하는 경우에만 변경을 주고 그렇지 않은 경우는 기본값이 적용되도록 수정하였습니다.

Related to: #73

ingbinsee committed:
🤝 [Merge] : 원격 저장소와 로컬 저장소 동기화

dani-day reviewed (approved):
추가 의견 없습니다. 수고하셨습니다! 👍 🥇
